pkg/pravega/pravega_segmentstore.go

```diff
@@ -116,13 +138,66 @@ func makeSegmentstorePodSpec(pravegaCluster *api.PravegaCluster) corev1.PodSpec
 				},
 				EnvFrom: environment,
 				VolumeMounts: []corev1.VolumeMount{
+					{
+						Name:      "podinfo",
+						MountPath: "/etc/podinfo",
+						ReadOnly:  false,
+					},
 					{
 						Name:      cacheVolumeName,
 						MountPath: cacheVolumeMountPoint,
 					},
 				},
 			},
 		},
+		Affinity: &corev1.Affinity{
+			PodAntiAffinity: &corev1.PodAntiAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+					{
+						LabelSelector: &metav1.LabelSelector{
+							MatchExpressions: []metav1.LabelSelectorRequirement{
+								{
+									Key:      "component",
+									Operator: metav1.LabelSelectorOpIn,
+									Values:   []string{"pravega-segmentstore"},
+								},
+								{
+									Key:      "pravega_cluster",
+									Operator: metav1.LabelSelectorOpIn,
+									Values:   []string{pravegaCluster.Name},
+								},
+							},
+						},
+						TopologyKey: "kubernetes.io/hostname",
+					},
+				},
+			},
+		},
+		Volumes: []corev1.Volume{
+			{
+				Name: "podinfo",
+				VolumeSource: corev1.VolumeSource{
+					DownwardAPI: &corev1.DownwardAPIVolumeSource{
```

Comment:
An surprised that you exposed the downward API as a file instead of as a set of environment variables.  Rationale?

Reply:
Updated. Downward API is now exposed via env vars.